apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,132 @@
+  word.charAt(0).toUpperCase() + word.slice(1);
+
+const getFormattedPath = (path: string, title: string) =>
+  `${path
+    .replace(/#.+$/, '')
+    .split('/')
+    .map(element => element.replaceAll('-', ' '))
+    .map(element => uppercaseFirst(element))
+    .filter(Boolean)
+    .join(' > ')} — ${title}`;
 
-export const SearchButton: FC = () => {
-  const [isOpen, setIsOpen] = useState(false);
+const SearchButton: FC = () => {
+  const { resolvedTheme } = useTheme();
   const t = useTranslations();
-  const openSearchBox = () => setIsOpen(true);
-  const closeSearchBox = () => setIsOpen(false);
-
-  useKeyboardCommands(cmd => {
-    switch (cmd) {
-      case 'cmd-k':
-        openSearchBox();
-        break;
-      case 'escape':
-        closeSearchBox();
-        break;
-      default:
-    }
-  });
-
-  const { os } = useDetectOS();
-
-  const osCommandKey = os === 'MAC' ? '⌘' : 'Ctrl';
-  const isOSLoading = os === 'LOADING';
+  const locale = useLocale();
+  const colorScheme = resolvedTheme as 'light' | 'dark';
+  const router = useRouter();
+
+  const sourceMap = {
+    title: 'pageSectionTitle',
+    description: 'formattedPath',
+    path: 'path',
+  };
+
+  const resultMap = {
+    ...sourceMap,
+    description: ({ path, pageSectionTitle }: ResultMapDescription) =>
+      getFormattedPath(path, pageSectionTitle),
+    path: ({ path, siteSection }: ResultMapPath) =>
+      siteSection.toLowerCase() === 'docs' ? `/${path}` : `/${locale}/${path}`,
+    section: 'siteSection',
+  };
 
   return (
     <>
-      <button
-        type="button"
-        onClick={openSearchBox}
-        className={styles.searchButton}
+      <OramaSearchButton
+        style={{ flexGrow: 1 }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
         aria-label={t('components.search.searchBox.placeholder')}
       >
-        <MagnifyingGlassIcon className={styles.magnifyingGlassIcon} />
-
         {t('components.search.searchBox.placeholder')}
-        <kbd
-          title={`${osCommandKey} K`}
-          className={classNames(styles.shortcutIndicator, {
-            'opacity-0': isOSLoading,
-          })}
-        >
-          <abbr>{osCommandKey} K</abbr>
-        </kbd>
-      </button>
-
-      {isOpen ? <WithSearchBox onClose={closeSearchBox} /> : null}
+      </OramaSearchButton>
+
+      <OramaSearchBox
+        index={{ api_key: ORAMA_CLOUD_API_KEY, endpoint: ORAMA_CLOUD_ENDPOINT }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
+        sourceBaseUrl={BASE_URL}
+        sourcesMap={sourceMap}
+        resultMap={resultMap}
+        facetProperty="siteSection"
+        linksTarget="_self"
+        highlightTitle={{
+          caseSensitive: false,
+          HTMLTag: 'b',
+          CSSClass: 'font-bold',
+        }}
+        searchParams={DEFAULT_ORAMA_QUERY_PARAMS}
+        suggestions={DEFAULT_ORAMA_SUGGESTIONS}
+        chatMarkdownLinkHref={({ href }) => {
+          if (!href) return href;
```

Comment:
Let's ensure we always use brackets on if statements.

Reply:
Thanks! Should we add it to lint rules?

Reply:
Feel free to do so 💯